infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -447,6 +491,7 @@ private void fetchTopologyFromClusterManager() {
      */
     private void stopLogReplication() {
         if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && isLeader.get()) {
+            isLeader.set(false);
```

Comment:
I think it is there in case it is stopped from processEvent on an exception (doesn't really imply we are leaders) or if called from processTopologyChangeNotification which also does not assume we're leaders.
You could probably set it right when it comes back from stopLogReplication in the processLockRelease, right? ... Cause imagine we get a processTopologyChangeNotification, in this case maybe we stop just because we are no longer active, but that does not mean we lost leadership.

Reply:
Ok makes sense